src/transformers/modeling_utils.py

```diff
@@ -1433,29 +1767,28 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         else:
             resolved_archive_file = None
 
+        # We'll need to download and cache each checkpoint shard if the checkpoint is sharded.
+        if is_sharded:
+            # resolved_archive_file becomes a list of files that point to the different checkpoint shards in this case.
+            resolved_archive_file, sharded_metadata = get_checkpoint_shard_files(
```

Comment:
This converts `resolved_archive_file` which is currently the index file into a list of state dict files (each file containing a shard of the checkpoint).